### PR TITLE
[Merged by Bors] - ET-4189 tenant specific elastic search handling

### PR DESCRIPTION
**Pull request by rustonaut**
Add tenant specific elastic search handling.

Should be merged into #872 (once both have been approved) and merged with that PR together into master.

Moves the  generic elastic search client into `web-api-shared`, it also needed to be slightly changed as it turns out it's design was still subtley a bit too much coupled with it's usage in `web-api`

**References:**

- merge into: #872 
- story: [ET-4322]
- issue: [ET-4189]

[ET-4322]: https://xainag.atlassian.net/browse/ET-4322?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[ET-4189]: https://xainag.atlassian.net/browse/ET-4189?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: web-api-db-ctrl/src/elastic.rs [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    Ok(())
+}
+
+pub(crate) async fn setup_legacy_tenant(
+    elastic: &Client,
+    default_index: &str,
+    tenant_id: &TenantId,
+) -> Result<(), Error> {
+    elastic
+        .with_index("_aliases")
```
Comment (rustonaut) [inline web-api-db-ctrl/src/elastic.rs:52]: It's a bit hacky but the cleanest solution without adding additional features to our ad-hoc elastic search client or using the elastic search library.

File: web-api-db-ctrl/elastic-search/mapping.json [hunk 2]
```diff
@@ -0,0 +1 @@
+../../web-api/elastic-search/mapping.json
```
Comment (rustonaut) [inline web-api-db-ctrl/elastic-search/mapping.json:1]: did github always display symlinks this way or is something broken?

After the update of the integration tests we will only have the mapping in `web-api-db-ctrl` and the one in `web-api` will go away as it and the create_es_index docker container are no longer needed.

File: web-api-db-ctrl/src/postgres.rs [hunk 3]
```diff
@@ -36,19 +36,19 @@ use crate::Error;
 static MT_USER: Lazy<QuotedIdentifier> = Lazy::new(|| "web-api-mt".parse().unwrap());
 
 static PUBLIC_SCHEMA_MIGRATOR: Lazy<Migrator> = Lazy::new(|| {
-    let mut migrator = sqlx::migrate!("migrations/public");
+    let mut migrator = sqlx::migrate!("postgres/public");
```
Comment (rustonaut) [inline web-api-db-ctrl/src/postgres.rs:39]: I changed the folder name for consistency with elastic search. 

Now schemas for elastic-search are under `elastic-search/` and schemas for postgres are under `postgres/`

**Review by janpetschexain — APPROVED** [review]
lgtm

**Comment by rustonaut** [comment]
bors merge

**Comment by rustonaut** [comment]
bors merge

**Pull request closed without merging**